### ISSUE TITLE
Layout not enforced for if(_)then_()else(_)

### ISSUE THREAD

**Issue by apblack** [gracelang/minigrace#221]
This compiles

```
if {funk>7} then {
    print "Good"
} 
else {
    print "not so good"
}
```

whereas the compiler ought to complain that there is no method `else`.

**Comment by KimBruce** [comment]
Do you really want to ban this?  There are a variety of understandable ways to indent if-then-else:

```
if (true) then {
    print "hello"
}
else
{
    print "goodbye"
}
```
or 
```
if (true) 
    then {print "hello"}
    else {print "goodbye"}

```
or
```
if (true) 
then {print "hello"}
else {print "goodbye"}
```
and finally, my favorite:
```
if (true) then {
    print "hello"
} else {
    print "goodbye"
}
```

All of these currently work.  While the next to last perhaps shouldn't work, I don't see the problem with the others.  In particular, the last example is what I use rather consistently.  However, I understand the above might violate our currently written rules.  Can they be rewritten (gracefully) to accommodate these?

**Comment by kjx** [comment]
On 11/10/2016, at 19:32PM, Kim Bruce <notifications@github.com> wrote:

if this is one request

> if (true) then {
>     print "hello"
> }
> else
> {
>     print "goodbye"
> }

(I'm assuming I'm right in guessing that "if" and "else" are indented the same... 

then isn't this one request 

print "foo"
print "bar" 

> if (true) 
> then {print "hello"}
> else {print "goodbye"}

ditto

J

**Comment by apblack** [comment]
This is fixed in commit 826f1578.  By "fixed" I mean that @KimBruce's examples 1 and 3 now generate static errors, while examples 2 and 4 continue to work.

Because the _minigrace_ parser deals with `if(_)then(_)else(_)` as a special case, the error message is also a special case, and says that the `then` part of an `if(_)then(_)…` must be indented more than the `if`.

The general case of the body of a block needing to be indented was also not being caught, as I discovered when I lectured to my students about the importance of indentation.  Since commit 0b6e223e40, this case is also reported as a syntax error.
